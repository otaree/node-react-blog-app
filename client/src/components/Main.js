import React from 'react';
import { Switch, Route } from 'react-router-dom';
import Blogs from './Blogs';
import Blog from './Blog';
import CreateBlog from './CreateBlog';
import EditBlog from './EditBlog';
import Auth from './Auth';

const Main = () => (
    <main>
        <Switch>
            <Route exact path='/' component={Blogs} /> 
            <Route path='/auth' component={Auth} /> 
            <Route path="/blogs/create" component={CreateBlog} />
            <Route path="/blogs/edit/:id" component={EditBlog} />
            <Route exact path="/blogs/:id" component={Blog} />
        </Switch>
    </main>
);

export default Main;