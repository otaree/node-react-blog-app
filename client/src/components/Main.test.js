import React from 'react';
import { shallow } from 'enzyme';
import Main from './Main';

describe('Main', () => {
    const main = shallow(<Main />);

    it('renders properly', () => {
        expect(main).toMatchSnapshot();
    });
});