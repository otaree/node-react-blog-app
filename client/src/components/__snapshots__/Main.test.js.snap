// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Switch>
        <Route
                component={[Function]}
                exact={true}
                path="/"
        />
        <Route
                component={[Function]}
                path="/auth"
        />
        <Route
                component={[Function]}
                path="/blogs/create"
        />
        <Route
                component={[Function]}
                path="/blogs/edit/:id"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/blogs/:id"
        />
</Switch>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            exact={true}
            path="/"
/>,
          <Route
            component={[Function]}
            path="/auth"
/>,
          <Route
            component={[Function]}
            path="/blogs/create"
/>,
          <Route
            component={[Function]}
            path="/blogs/edit/:id"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/blogs/:id"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/auth",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/blogs/create",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/blogs/edit/:id",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/blogs/:id",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Switch>
          <Route
                    component={[Function]}
                    exact={true}
                    path="/"
          />
          <Route
                    component={[Function]}
                    path="/auth"
          />
          <Route
                    component={[Function]}
                    path="/blogs/create"
          />
          <Route
                    component={[Function]}
                    path="/blogs/edit/:id"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/blogs/:id"
          />
</Switch>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
/>,
            <Route
              component={[Function]}
              path="/auth"
/>,
            <Route
              component={[Function]}
              path="/blogs/create"
/>,
            <Route
              component={[Function]}
              path="/blogs/edit/:id"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/blogs/:id"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/auth",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/blogs/create",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/blogs/edit/:id",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/blogs/:id",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
